File: BioJulia/count.jl
# Benchmark the counting of alanine residues in a PDB file

using BioStructures

pdb_filepath = "pdbs/1AKE.pdb"
struc = read(pdb_filepath, PDB)

function counter()
    alanineselector(res::AbstractResidue) = resnameselector(res, ["ALA"])
    return countresidues(struc, alanineselector)
end

# Run to JIT compile
counter()

elapsed = @elapsed counter()

println(elapsed)

File: BioRuby/count.rb
# Benchmark the counting of alanine residues in a PDB file

require "bio"
require "benchmark"

pdb_filepath = "pdbs/1AKE.pdb"
pdb = Bio::PDB.new(File.read(pdb_filepath))

elapsed = Benchmark.realtime {
    pdb.find_residue { |res| res.resName == "ALA" }.length
}

print elapsed, "\n"
